src/create-keyed-resource.test.js

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import { render, waitForElement } from "react-testing-library";
+import { identity } from "ramda";
+import useEntry from "./use-entry";
+import RemoteResourceBoundary from "./RemoteResourceBoundary";
+import createKeyedResource from "./create-keyed-resource";
+
+// ---------------------------
+// Mocks
+// ---------------------------
+
+const MockResourceConsumer = ({ resource, index }) => {
+  const [entry] = useEntry(resource, [index]);
+
+  return entry;
+};
+
+// ---------------------------
+// Tests
+// ---------------------------
+
+describe("createKeyedResource", () => {
+  it("creates a resource", async () => {
+    const resource = createKeyedResource(identity, () =>
+      Promise.resolve("resolved")
+    );
+
+    const { getByText } = render(
+      <RemoteResourceBoundary
+        fallback={<p>Loading...</p>}
+        renderError={() => <p>error</p>}
+      >
+        <MockResourceConsumer resource={resource} index={0} />
+      </RemoteResourceBoundary>
+    );
+
+    await waitForElement(() => getByText("Loading..."));
+    await waitForElement(() => getByText("resolved"));
```

Comment:
It seems like this test doesn't quite match up with the description. If we're validating that `createKeyedResource` creates a resource should we test that the shape of the `resource` is correct instead?

Reply:
That is a really good point!